nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -64,6 +64,7 @@ namespace nav2_costmap_2d
 void ObstacleLayer::onInitialize()
 {
   node_->get_parameter_or_set(name_ + "." + "enabled", enabled_, true);
+  node_->get_parameter_or_set(name_ + "." + "enable_always", local_costmap_flag_, true);
```

Comment:
You should change the name of "local_costmap_flag" to "enable_always" since that's the way you're using it internally, and it's more clear.